src/dsp.jl

```diff
@@ -410,3 +411,64 @@ function mfilter(r, s)
   n = length(r) - 1
   sfilt(f, padded(s, (0, n)))[n+1:end]
 end
+
+"""
+$(SIGNATURES)
+Inverse Short Time Fourier Transform (ISTFT). 
+
+The signal windowing must obey the constraint of "nonzero overlap add" (NOLA). 
+Implementation based on Hristo (2019, 2020) and `istft` in `scipy`. 
+
+H. Zhivomirov. On the Development of STFT-analysis and ISTFT-synthesis Routines 
+and their Practical Implementation. TEM Journal, ISSN: 2217-8309, DOI: 10.18421/TEM81-07, 
+Vol. 8, No. 1, pp. 56-64, Feb. 2019. 
+(http://www.temjournal.com/content/81/TEMJournalFebruary2019_56_64.pdf)
+
+Hristo Zhivomirov (2020). Inverse Short-Time Fourier Transform (ISTFT) with Matlab 
+(https://www.mathworks.com/matlabcentral/fileexchange/45577-inverse-short-time-fourier-transform-istft-with-matlab), 
+MATLAB Central File Exchange. Retrieved October 21, 2020. 
+"""
+function istft(X::AbstractMatrix{Complex{T}}, 
+               n::Int, 
+               noverlap::Int; 
+               onesided::Bool=true, 
+               window::Union{Function,AbstractVector,Nothing}=nothing) where {T<:AbstractFloat}
+  (window === nothing) && (window = rect)
+  win, norm2 = Periodograms.compute_window(window, n)
+
+  nstep = n - noverlap
+  nseg = size(X, 2)
+  outputlength = n + (nseg-1) * nstep
+
+  iX = onesided ? irfft(X, n, 1) : ifft(X, 1)
+  iX .*= win
+  x = zeros(eltype(iX), outputlength)
+  normw = zeros(T, outputlength)
+  for i = 1:nseg
+      x[1+(i-1)*nstep:n+(i-1)*nstep] += iX[:,i]
+      normw[1+(i-1)*nstep:n+(i-1)*nstep] += win .^ 2
+  end
+
+  (sum(normw[n÷2:end-n÷2] .> 1e-10) != length(normw[n÷2:end-n÷2])) && (
+      @warn "NOLA condition failed, STFT may not be invertible")
+  x .*= nstep/norm2
+  x
+end
+
+"""
+$(SIGNATURES)
+Spectral whitening/flattening in the frequency domain.
+
+M. W. Lee, Spectral whitening in the frequency domain, 1986.
```

Comment:
Unclear from the comment that this is a reference. Use consistent style as used elsewhere in the package?

Reply:
May I know which function in the package I can refer to for the reference style?

Reply:
Currently I have included references only in the code, not the docs:
https://github.com/org-arl/SignalAnalysis.jl/blob/94b235a85b9c5be16a84c4782cea00a2d54c55aa/src/tfa.jl#L73
but I'm open to us changing that throughout and including in docs if you think we should do that. I just would like to be consistent throughout.

Reply:
I've been looking around for the standard reference style in Julia but couldn't find something satisfying. The references of the two functions are modified to be consistent with the existing references, e,g,, # author name, paper, volume/page, year.